src/service-module/mutations.js

```diff
@@ -35,10 +35,19 @@ export default function makeServiceMutations (servicePath, { debug, globalModels
   }
 
   function updateItems (state, items) {
-    const { idField, replaceItems, addOnUpsert } = state
+    const { idField, replaceItems, addOnUpsert, customMerge } = state
```

Comment:
so like a `updateUserIfMatch(item)` and then internally compare ids or what-not

Reply:
Yeah.  Something wrapped into its own happy place.  Please also check out @rayfoss's PR to see if there's anything required to play nice with that.

Reply:
This change from today may also be important: https://github.com/feathers-plus/feathers-vuex/issues/184

Reply:
I agree with putting it in a mutator, it looks a bit nasty the way it is now. 

Seeming as this will basically involve rewriting, might be worth at this point closing this PR, push all the other new stuff, and then implement it again in a mutator and accounting for the other stuff.

Adding custom merge should probably be in it's own PR along with unit tests and docs. 
Fixing the get action should be bundled with removing the setCurrents ( #186 )  as they are both potentially breaking changes.